android/java/org/chromium/chrome/browser/toolbar/top/BraveToolbarLayoutImpl.java

```diff
@@ -441,11 +441,14 @@ public void run() {
         mTabModelSelectorTabModelObserver = new TabModelSelectorTabModelObserver(selector) {
             @Override
             public void didSelectTab(Tab tab, @TabSelectionType int type, int lastId) {
-                if (getToolbarDataProvider().getTab() == tab && mBraveRewardsNativeWorker != null
-                        && !tab.isIncognito()) {
+                if (mBraveRewardsNativeWorker != null && !tab.isIncognito()) {
                     mBraveRewardsNativeWorker.OnNotifyFrontTabUrlChanged(
                             tab.getId(), tab.getUrl().getSpec());
-                    showWalletIcon(mTabsWithWalletIcon.contains(tab.getId()));
+                    if (getToolbarDataProvider().getTab() == tab) {
+                        showWalletIcon(mTabsWithWalletIcon.contains(tab.getId()));
+                    } else {
+                        mWalletLayout.setVisibility(mTabsWithWalletIcon.contains(tab.getId()) ? View.VISIBLE: View.GONE);
```

Comment:
This needs to be formatted. Always run `npm run lint` to fix the formatting issues before pushing the changes, it can cause CI failure as well.

Reply:
I run it but no effect, I also formatted code snippet in Android Studio... I think I should format by copying whole file and formatting it in android studio